src/rgw/rgw_rest_oidc_provider.cc

```diff
@@ -46,52 +46,97 @@ void RGWRestOIDCProvider::send_response()
 }
 
 
+static std::string format_creation_date(ceph::real_time now)
+{
+  struct timeval tv;
+  real_clock::to_timeval(now, tv);
+
+  struct tm result;
+  gmtime_r(&tv.tv_sec, &result);
+  char buf[30];
+  strftime(buf,30,"%Y-%m-%dT%H:%M:%S", &result);
+  sprintf(buf + strlen(buf),".%03dZ",(int)tv.tv_usec/1000);
+  return buf;
+}
+
+
 RGWCreateOIDCProvider::RGWCreateOIDCProvider()
   : RGWRestOIDCProvider(rgw::IAM::iamCreateOIDCProvider, RGW_CAP_WRITE)
 {
 }
 
+inline constexpr int MAX_OIDC_NUM_CLIENT_IDS = 100;
+inline constexpr int MAX_OIDC_CLIENT_ID_LEN = 255;
+inline constexpr int MAX_OIDC_NUM_THUMBPRINTS = 5;
+inline constexpr int MAX_OIDC_THUMBPRINT_LEN = 40;
+inline constexpr int MAX_OIDC_URL_LEN = 255;
+
 int RGWCreateOIDCProvider::init_processing(optional_yield y)
 {
-  provider_url = s->info.args.get("Url");
-  if (provider_url.empty()) {
+  info.provider_url = s->info.args.get("Url");
+  if (info.provider_url.empty()) {
     s->err.message = "Missing required element Url";
     return -EINVAL;
   }
+  if (info.provider_url.size() > MAX_OIDC_URL_LEN) {
+    s->err.message = "Url cannot exceed the maximum length of "
+        + std::to_string(MAX_OIDC_URL_LEN);
+    return -EINVAL;
+  }
 
   auto val_map = s->info.args.get_params();
   for (auto& it : val_map) {
-      if (it.first.find("ClientIDList.member.") != string::npos) {
-          client_ids.emplace_back(it.second);
+    if (it.first.find("ClientIDList.member.") != string::npos) {
+      if (it.second.size() > MAX_OIDC_CLIENT_ID_LEN) {
+        s->err.message = "ClientID cannot exceed the maximum length of "
+            + std::to_string(MAX_OIDC_CLIENT_ID_LEN);
+        return -EINVAL;
       }
-      if (it.first.find("ThumbprintList.member.") != string::npos) {
-          thumbprints.emplace_back(it.second);
+      info.client_ids.emplace_back(it.second);
+    }
+    if (it.first.find("ThumbprintList.member.") != string::npos) {
+      if (it.second.size() > MAX_OIDC_THUMBPRINT_LEN) {
+        s->err.message = "Thumbprint cannot exceed the maximum length of "
+            + std::to_string(MAX_OIDC_THUMBPRINT_LEN);
+        return -EINVAL;
       }
+      info.thumbprints.emplace_back(it.second);
+    }
   }
 
-  if (thumbprints.empty()) {
+  if (info.thumbprints.empty()) {
     s->err.message = "Missing required element ThumbprintList";
     return -EINVAL;
   }
+  if (info.thumbprints.size() > MAX_OIDC_NUM_THUMBPRINTS) {
+    s->err.message = "ThumbprintList cannot exceed the maximum size of "
+        + std::to_string(MAX_OIDC_NUM_THUMBPRINTS);
+    return -EINVAL;
+  }
+
+  if (info.client_ids.size() > MAX_OIDC_NUM_CLIENT_IDS) {
+    s->err.message = "ClientIDList cannot exceed the maximum size of "
+        + std::to_string(MAX_OIDC_NUM_CLIENT_IDS);
+    return -EINVAL;
+  }
+
+  info.tenant = s->user->get_tenant();
```

Comment:
I have this generic comment - what about account id/ name associated with the OIDC Provider?